src/shared/bus-util.h

```diff
@@ -23,11 +23,31 @@ typedef enum BusTransport {
 } BusTransport;
 
 typedef struct BusLocator {
-        const char    *destination;
-        const char    *path;
-        const char    *interface;
+        const char *destination;
+        const char *path;
+        const char *interface;
 } BusLocator;
 
+typedef struct BusObjectImplementation BusObjectImplementation;
+
+typedef struct BusObjectVtablePair {
+        const sd_bus_vtable *vtable;
+        sd_bus_object_find_t object_find;
+} BusObjectVtablePair;
+
+struct BusObjectImplementation {
+        const char *path;
+        const char *interface;
+        const BusObjectVtablePair *vtables;
```

Comment:
i'd do two separate fields here, one for the plain vtables and one for the fallbcak vtables, see earlier discussion